File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/diagnostics/SKException.java
// Copyright (c) Microsoft. All rights reserved.
package com.microsoft.semantickernel.diagnostics;

import javax.annotation.Nullable;

/** Provides the base exception from which all Semantic Kernel exceptions derive. */
public class SKException extends RuntimeException {

    /** Initializes a new instance of the {@code SKException} class with a default message. */
    protected SKException() {
        super();
    }

    /**
     * Initializes a new instance of the {@code SKException} class with its message set to {@code
     * message}.
     *
     * @param message A string that describes the error.
     */
    protected SKException(@Nullable String message) {
        super(message);
    }

    /**
     * Initializes a new instance of the {@code SKException} class with its message set to {@code
     * message}.
     *
     * @param message A string that describes the error.
     * @param cause The exception that is the cause of the current exception.
     */
    protected SKException(@Nullable String message, @Nullable Throwable cause) {
        super(message, cause);
    }
}

File: dotnet/src/SemanticKernel.Abstractions/Diagnostics/SKException.cs
// Copyright (c) Microsoft. All rights reserved.

using System;

namespace Microsoft.SemanticKernel.Diagnostics;

/// <summary>
/// Represents the base exception from which all Semantic Kernel exceptions derive.
/// </summary>
public class SKException : Exception
{
    /// <summary>
    /// Initializes a new instance of the <see cref="SKException"/> class.
    /// </summary>
    public SKException()
    {
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="SKException"/> class with a specified error message.
    /// </summary>
    /// <param name="message">The error message that explains the reason for the exception.</param>
    public SKException(string? message) : base(message)
    {
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="SKException"/> class with a specified error message and a reference to the inner exception that is the cause of this exception.
    /// </summary>
    /// <param name="message">The error message that explains the reason for the exception.</param>
    /// <param name="innerException">The exception that is the cause of the current exception, or a null reference if no inner exception is specified.</param>
    public SKException(string? message, Exception? innerException) : base(message, innerException)
    {
    }
}
